### PR TITLE
fix(Select): fix select options positioning

### PR DESCRIPTION
**Pull request by pladaria**
fixes: https://tinyurl.com/2zqhm4or

This new approach uses a fixed positioning inside a portal

### CODE REVIEW

File: src/select.tsx [hunk 1]
```diff
@@ -408,56 +403,58 @@ const Select: React.FC<SelectProps> = ({
                     }}
                     disableScroll
                 >
```
Comment (pladaria) [inline src/select.tsx:405]: review this ignoring whitespace:
https://github.com/Telefonica/mistica-web/pull/802/files?diff=unified&w=1

File: src/select.tsx [hunk 2]
```diff
@@ -107,16 +108,10 @@ const Select: React.FC<SelectProps> = ({
                 const MARGIN_TOP_SIZE = 12;
                 const PADDING_SIZE = 16;
 
-                // get a possible non-static parent, like a modal dialog
-                const nonStaticParentElement = focusableRef.current?.offsetParent;
-                const parentElementRect = nonStaticParentElement?.getBoundingClientRect() || {
-                    left: 0,
-                    top: 0,
-                };
```
Comment (pladaria) [inline src/select.tsx:115]: this approach had issues with other non static parent combinations

**Review by yceballost — APPROVED** [review]

File: src/__screenshot_tests__/__image_snapshots__/dialog-screenshot-test-tsx-select-options-are-correctly-positioned-inside-a-dialog-1-snap.png
Comment (marcoskolodny) [inline src/__screenshot_tests__/__image_snapshots__/dialog-screenshot-test-tsx-select-options-are-correctly-positioned-inside-a-dialog-1-snap.png:1]: what changed here?

File: src/select.css.ts [hunk 3]
```diff
@@ -17,9 +17,12 @@ export const vars = {
     maxHeight,
 };
 
+/** Must be equal or higher than the dialog's z-index */
+const OPTIONS_ZINDEX = 26;
```
Comment (atabel) [inline src/select.css.ts:21]: lol

**Review by atabel — APPROVED** [review]

File: src/__screenshot_tests__/__image_snapshots__/dialog-screenshot-test-tsx-select-options-are-correctly-positioned-inside-a-dialog-1-snap.png
Comment (pladaria) [reply src/__screenshot_tests__/__image_snapshots__/dialog-screenshot-test-tsx-select-options-are-correctly-positioned-inside-a-dialog-1-snap.png:1]: yep, it looks like the shadow is slightly different. I don't know why :/

**Review by marcoskolodny — APPROVED** [review]

**Pull request merged**